README.rst

```diff
@@ -59,6 +59,89 @@ Next, let's parse something and get an abstract syntax tree:
 You'd typically then use a ``nodes.NodeVisitor`` subclass (see below) to walk
 the tree and do something useful with it.
 
+Another example would be to implement a parser for ``.ini``-files. Consider the following:
+
+.. code:: python
+
+    grammar = Grammar(
+        r"""
+        expr        = (entry / emptyline)*
+        entry       = section pair*
+
+        section     = lpar word rpar ws
+        pair        = key equal value ws?
+
+        key         = word+
+        value       = (word / quoted)+
+        word        = ~r"[-\w]+"
+        quoted      = ~'"[^\"]+"'
+        equal       = ws? "=" ws?
+        lpar        = "["
+        rpar        = "]"
+        ws          = ~"\s*"
+        emptyline   = ws+
+        """
+    )
+
+
+We could now implement a subclass of `NodeVisitor` like so:
+
+.. code:: python
+
+    class IniVisitor(NodeVisitor):
+        def visit_expr(self, node, visited_children):
+            """ Returns the overall output. """
+            output = {}
+            for child in visited_children:
+                output.update(child[0])
+            return output
+
+        def visit_entry(self, node, visited_children):
+            """ Makes a dict of the section (as key) and the key/value pairs. """
+            return {visited_children[0]: dict(visited_children[1])}
```

Comment:
This looks good! Since it's documentation, I might try to be screechingly obvious here and do something like `key, value = visited_children`. I'll leave it to you if you want to do a 2nd patch.